odoo/tools/misc.py

```diff
@@ -1208,6 +1208,25 @@ def format_date(env, value, lang_code=False, date_format=False):
 
     return babel.dates.format_date(value, format=date_format, locale=locale)
 
+def parse_date(env, value, lang_code=False):
+    '''
+        Parse the date from a given format. If it is not a valid format for the
+        localization, return the original string.
+
+        :param env: an environment.
```

Comment:
Maybe specify that it's an *odoo* environment (assuming that's one) e.g.
```suggestion
        :param env: an :class:`~odoo.api.Environment`.
```
?